src/Quadrature.cpp

```diff
@@ -0,0 +1,1305 @@
+ */
+void TYPolarQuad::setNumPolarAngles(const int num_polar) {
+
+  if (num_polar > 6)
+    log_printf(ERROR, "Unable to set the number of polar angles to %d "
+               "for TYPolarQuad (max 6 angles)", num_polar);
+
+  Quadrature::setNumPolarAngles(num_polar);
+}
+
+
+/**
+ * @brief Routine to initialize the polar quadrature.
+ * @details This routine uses the tabulated values for the Tabuchi-Yamamoto
+ *          polar angle quadrature, including the sine thetas and weights.
+ */
+void TYPolarQuad::initialize() {
+
+  /* Call parent class initialize routine */
+  Quadrature::initialize();
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  double* thetas = new double[_num_polar/2*_num_azim/4];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Tabuchi-Yamamoto polar angle quadrature */
+  if (_num_polar == 2) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a] = asin(0.798184);
+    }
+  }
+
+  else if (_num_polar == 4) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = asin(0.363900);
+      thetas[a*(_num_polar/2)+1] = asin(0.899900);
+    }
+  }
+
+  else if (_num_polar == 6) {
+    for (int a=0; a < _num_azim/4; a++) {
+      thetas[a*(_num_polar/2)] = asin(0.166648);
+      thetas[a*(_num_polar/2)+1] = asin(0.537707);
+      thetas[a*(_num_polar/2)+2] = asin(0.932954);
+    }
+  }
+
+  /* Set the arrays of thetas */
+  Quadrature::setThetas(thetas, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] thetas;
+}
+
+
+/**
+ * @brief Calculates total weights for every azimuthal/polar combination based
+ *        on the TY quadrature
+ * @param solve_3D Boolean indicating whether this is a 3D quadrature
+ */
+void TYPolarQuad::precomputeWeights(bool solve_3D) {
+
+  /* Allocate temporary arrays for tabulated quadrature values */
+  FP_PRECISION* weights = new FP_PRECISION[_num_polar/2*_num_azim/4];
+
+  /* Tabulated values for the sine thetas and weights for the
+   * Tabuchi-Yamamoto polar angle quadrature */
+  if (_num_polar == 2) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a] = 0.5;
+    }
+  }
+
+  else if (_num_polar == 4) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.212854 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.787146 / 2.0;
+    }
+  }
+
+  else if (_num_polar == 6) {
+    for (int a=0; a < _num_azim/4; a++) {
+      weights[a*(_num_polar/2)] = 0.046233 / 2.0;
+      weights[a*(_num_polar/2)+1] = 0.283619 / 2.0;
+      weights[a*(_num_polar/2)+2] = 0.670148 / 2.0;
+    }
+  }
+
+  /* Set the arrays of sin thetas and weights */
+  Quadrature::setPolarWeights(weights, _num_polar/2*_num_azim/4);
+
+  /* Deallocate temporary arrays */
+  delete [] weights;
+
+  Quadrature::precomputeWeights(solve_3D);
+}
+
+
+
+/**
+ * @brief Dummy constructor calls the parent constructor.
+ */
+LeonardPolarQuad::LeonardPolarQuad(): Quadrature() {
+  _quad_type = LEONARD;
```

Comment:
Perhaps this should have a default for `_num_polar`? Otherwise the `LeonardPolarQuad::initialize()` method below will pass without doing anything right?